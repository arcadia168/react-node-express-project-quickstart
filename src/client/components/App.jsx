import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types'

const HelloWorld = () => <h1 data-testid="app-title" className="app__title">Hello World</h1>;

const App = () => {
  return (
    <Router>
      <Route exact path="/" component={HelloWorld} />
      <Button>Hello React Start with Bootstrap!</Button>
    </Router>
  )
}

App.propTypes = {
  axiosInstance: PropTypes.func,
}

export default App;
