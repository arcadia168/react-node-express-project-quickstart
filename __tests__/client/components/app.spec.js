import React from 'react';
import App from '../../../src/client/components/app';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import path from 'path';
import fs from 'fs';

// const pathName = path.resolve(__dirname, `../../__mocks__/validCardResults.json`);
// const mockCards = fs.readFileSync(pathName, 'utf8');

describe('App', () => {
    const render = customProps => {
        const props = {
            // Default props
            ...customProps,
        }
        return mount(<App {...props}/>);
    }

    it('renders the app as expected', () => {
        const component = renderer.create(
            <App />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('successfully renders the app', () => {
        const renderedApp = render();
        expect(
            renderedApp.find('[data-testid="app-title"]').text(),
        ).toBe('Hello World');
    });

    it('Assigns an object to the prop axiosInstance when passed', () => {
        const axiosInstanceMock = jest.fn();
        const renderedApp = render({ axiosInstance: axiosInstanceMock});
        expect(renderedApp.props().axiosInstance).toEqual(axiosInstanceMock);
    });
});

