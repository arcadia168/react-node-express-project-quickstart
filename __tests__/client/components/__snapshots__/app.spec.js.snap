// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app as expected 1`] = `
Array [
  <h1
    className="app__title"
    data-testid="app-title"
  >
    Hello World
  </h1>,
  <button
    className="btn btn-primary"
    disabled={false}
    type="button"
  >
    Hello React Start with Bootstrap!
  </button>,
]
`;
